### PR TITLE
Added distributed loading of PyTorch DiskDataset

### PR DESCRIPTION
**Pull request by arunppsg**
## Description

Added support for loading disk dataset in multiple processes.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by vsaravind01 — COMMENTED** [review]
The changes look good to me.

File: deepchem/data/pytorch_datasets.py [hunk 1]
```diff
@@ -87,16 +88,25 @@ def __init__(self,
...
+    def __len__(self):
+        return len(self.disk_dataset)
+
     def __iter__(self):
         worker_info = torch.utils.data.get_worker_info()
         n_shards = self.disk_dataset.get_number_shards()
         if worker_info is None:
-            first_shard = 0
-            last_shard = n_shards
+            process_id = 0
```
Comment (rbharath) [inline deepchem/data/pytorch_datasets.py:98]: @arunppsg Can we add a unit test for the changed functionality?

**Review by rbharath — COMMENTED** [review]
@arunppsg Is there any risk for breakage with these changes? What code uses TorchDiskDataset already?

**Comment by arunppsg** [comment]
> @arunppsg Is there any risk for breakage with these changes? What code uses TorchDiskDataset already?

`DiskDataset.make_pytorch_dataset` uses TorchDiskDataset method to make an iterable pytorch dataset. In these code snippets, we are only updating the process of distributing the shards across multiple workers in a single machine, so that each gpu gets a different copy of the dataset.

File: deepchem/data/pytorch_datasets.py (hunk 1 above)
Comment (rbharath) [reply deepchem/data/pytorch_datasets.py:98]: Documenting offline discussion, this is proving difficult to write. The existing unit tests pass. Will come back with additional tests in future PR

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**